packages/blocks/src/api/parser.js

```diff
@@ -394,54 +398,61 @@ export function getMigratedBlock( block ) {
  * @return {?Object} An initialized block object (if possible).
  */
 export function createBlockWithFallback( blockNode ) {
+	const { blockName: originalName } = blockNode;
 	let {
-		blockName: name,
 		attrs: attributes,
 		innerBlocks = [],
 		innerHTML,
 	} = blockNode;
+	const freeformContentFallbackBlock = getFreeformContentHandlerName();
+	const unregisteredFallbackBlock = getUnregisteredTypeHandlerName() || freeformContentFallbackBlock;
 
 	attributes = attributes || {};
 
 	// Trim content to avoid creation of intermediary freeform segments.
-	innerHTML = innerHTML.trim();
+	const originalUndelimitedContent = innerHTML = innerHTML.trim();
```

Comment:
This is [too clever](https://softwareengineering.stackexchange.com/questions/25276/why-is-cleverness-considered-harmful-in-programming-by-some-people). I'm still not totally sure I understand why / what we're doing with this line.

Reply:
Thanks, @aduth. The intent wasn't cleverness, but I see what you mean. We want `innerHTML` to be trimmed, and saving its value at that point is a separate concern.

The missing block needs the original undelimited content for determining whether the unregistered block has actual HTML content, for rendering a preview of the block HTML, and for converting to an HTML block. `originalUndelimitedContent` exists because [this pre-existing line](https://github.com/WordPress/gutenberg/pull/8274/files/79222f46ab26c11f4f6b8e8289b4ad811a2faf76#diff-7dae7b3890cf27659680c926b6992e31R438) changes the value of `innerHTML` to be the delimited HTML content. The delimited content is used in saving the value of the block.

The above approach, which I am responsible for, is a bit weird IMO. It would be more straightforward to:
1. Get rid of `originalUndelimitedContent`.
1. Stop overwriting `innerHTML` with the delimited content.
1. Add an `originalAttributes` attribute to `core/missing`.
1. Update `core/missing` to call `getCommentDelimitedContent` with `originalName`, `originalAttributes` and `originalContent` to create the save content.

If you agree this is cleaner, I can create a follow-up PR to make the change.

Reply:
I actually don't have much of a problem with `originalUndelimitedContent`, I just think we could be clearer / more verbose with how we assign / document its intent.

Also, I think with this type of assignment we're inadvertently assigning `innerHTML` as global outside the scope of the function:

![image](https://user-images.githubusercontent.com/1779930/47032632-d49d7080-d140-11e8-8b87-fffd553e7c7c.png)

Reply:
> I actually don't have much of a problem with originalUndelimitedContent, I just think we could be clearer / more verbose with how we assign it / document its intent.

✅ thanks
I don't love `originalUndelimitedContent` but plan to leave it after documenting its purpose and separating the assignment.

I don't believe this assignment should be creating a global since `innerHTML` is declared in the function scope [here](https://github.com/WordPress/gutenberg/pull/8274/files/79222f46ab26c11f4f6b8e8289b4ad811a2faf76#diff-7dae7b3890cf27659680c926b6992e31R405).